### DISCUSSION TITLE
Write logging output to a telnet port

### DISCUSSION THREAD

**Discussion by probonopd** [rsta2/circle#506] [Ideas]
We are currently using `#include <circle/logger.h>` in MiniDexed.
However, some devices running MiniDexed firmware don't even expose the HDMI port.

So we would like to output the log over the network, e.g., if a client is connected via telnet.

Is this possible?

**Comment by rsta2** [comment]
Should be possible via telnet, but requires some custom programming. The recommended way to transfer log messages over a local network in Circle is the syslog protocol, which is already implemented and shown in the [sample/33-syslog](https://github.com/rsta2/circle/tree/master/sample/33-syslog). See the README for details. The sample comes with a simple syslog server for Linux, but there are also apps for smartphones, with which you can receive and view the log messages.

**Reply by probonopd** [reply]
Thanks @rsta2. Will use that then. I have it working in principle, but am [struggling](https://github.com/probonopd/MiniDexed/pull/747#issuecomment-2481197825) with the hardcoded IP address of the server.

**Reply by rsta2** [reply]
Something like this should work (not tested):

```
// in config.h

{
...
	CIPAddress GetNetworkSyslogServerIPAddress (void);

private:
	CIPAddress m_SyslogServerIP;
...
}

// in config.cpp

void CConfig::Load (void)
{
...
	const u8 *pSyslogServerIP = m_Properties.GetIPAddress ("NetworkSyslogServerIPAddress");
	if (pSyslogServerIP)
	{
		m_SyslogServerIP.Set (pSyslogServerIP);
	}
...
}

CIPAddress CConfig::GetNetworkSyslogServerIPAddress (void)
{
	return m_SyslogServerIP;
}

// in network initialization:

{
...
	CIPAddress ServerIP = m_pConfig->GetNetworkSyslogServerIPAddress();
	if (ServerIP.IsSet () && !ServerIP.IsNull ())
	{
		static const u16 usServerPort = 8514;	// standard port is 514
		CString IPString;
		ServerIP.Format (&IPString);
		LOGNOTE ("Sending log messages to %s:%u",
			 (const char *) IPString, (unsigned) usServerPort);

		new CSysLogDaemon (m_pNet, ServerIP, usServerPort);
	}
...
}
```

To disable the syslog server, you only need to out-comment the line in *MiniDexed.ini* or set it to 0.0.0.0:

```
#NetworkSyslogServerIPAddress=192.168.0.143

NetworkSyslogServerIPAddress=0.0.0.0
```

**Reply by probonopd** [reply]
Thanks. Works.
